upup/models/cloudup/resources/addons/dns-controller.addons.k8s.io/k8s-1.12.yaml.template

```diff
@@ -36,15 +35,16 @@ spec:
 {{ range $arg := DnsControllerArgv }}
         - "{{ $arg }}"
 {{ end }}
-{{- if .EgressProxy }}
         env:
+        - name: KUBERNETES_SERVICE_HOST
+          value: "127.0.0.1"
+{{- if .EgressProxy }}
 {{ range $name, $value := ProxyEnv }}
```

Comment:
Now you've refactored this, I'm wondering if we need the "if .EgressProxy" guard.  For another PR thought!

Reply:
I moved the `if` below to add the new environment variable by default. There was actually a bug here, since if using DigitalOcean and EgressProxy together a duplicated `env:` key would be rendered.
Given the `range` on `ProxyEnv`, I'm not sure we need the `if` either, but didn't want to change this right now, I'm not familiar with this feature.